RAINCATCH-451: Update to use new simple store module

JameelB opened the pull request:
## What
Added missing id field in create handler to be used with the new simple store module. Added function to check for and remove angular generated variables. 

## JIRA Ticket
https://issues.jboss.org/browse/RAINCATCH-451

vchepeli commented on lib/server.js:
```diff
@@ -1,8 +1,22 @@
+  q = require('q'),
+  shortid = require('shortid');
+
+function cleanData(data) {
```
can you add some docs why we need this function

JameelB commented:
Moved addition of id field to each raincatcatcher modules(message, workflow, workorder, result). Moved removal of angular generated variables to demo-portal.
